### PR TITLE
Update support drawing for thrill rides and water rides

### PR DESCRIPTION
**Pull request by Gymnasiast**
Another chunk that should be easy to test.

### CODE REVIEW

**Review by karst — APPROVED** [review]
Seems all to be in working order
image

File: src/openrct2/ride/water/SplashBoats.cpp [hunk 1]
```diff
@@ -810,14 +826,19 @@ static void PaintSplashBoatsTrackLeftQuarterTurn5Tiles(
 
     if (trackSequence != 1 && trackSequence != 4)
     {
-        static constexpr uint8_t supportTypes[][7] = {
-            { 0, 0xFF, 5, 3, 0xFF, 5, 1 },
-            { 1, 0xFF, 2, 4, 0xFF, 2, 0 },
-            { 0, 0xFF, 3, 5, 0xFF, 3, 1 },
-            { 1, 0xFF, 4, 2, 0xFF, 4, 0 },
+        static constexpr std::optional<WoodenSupportSubType> supportTypes[NumOrthogonalDirections][7] = {
```
Comment (ZehMatt) [inline src/openrct2/ride/water/SplashBoats.cpp:829]: This is a bit of an odd one as it never asks if it holds a value or not, it's probably better to have a `None/Invalid` value for the gaps.

File: src/openrct2/ride/water/SplashBoats.cpp [hunk 2]
```diff
@@ -889,14 +910,19 @@ static void PaintSplashBoatsTrackRightQuarterTurn5Tiles(
 
     if (trackSequence != 1 && trackSequence != 4)
     {
-        static constexpr uint8_t supportTypes[][7] = {
-            { 0, 0xFF, 4, 2, 0xFF, 4, 1 },
-            { 1, 0xFF, 5, 3, 0xFF, 5, 0 },
-            { 0, 0xFF, 2, 4, 0xFF, 2, 1 },
-            { 1, 0xFF, 3, 5, 0xFF, 3, 0 },
+        static constexpr std::optional<WoodenSupportSubType> supportTypes[][7] = {
```
Comment (ZehMatt) [inline src/openrct2/ride/water/SplashBoats.cpp:913]: Same as above.

**Review by ZehMatt — CHANGES_REQUESTED** [review]
See comments

**Review by ZehMatt — APPROVED** [review]
LGTM, thanks.

**Comment by tupaschoal** [comment]
Argh, I squashed instead of merging, sorry!

**Comment by Gymnasiast** [comment]
Ah well, shit happens. Don’t worry too much about it.

**Pull request merged**